src/clinical/clinical-service.ts

```diff
@@ -324,14 +334,69 @@ export const getDonorSubmissionErrorUpdates = async (
     submitterDonorIds: [],
   };
 
-  await donorDao.findByPaginatedProgramId(programId, errorQuery).then(donorData => {
-    validDonors = donorData.donors
-      .filter(donor => donor.schemaMetadata.isValid)
-      .map(({ donorId }) => Number(donorId));
-  });
+  const donorData = (await donorDao.findByPaginatedProgramId(programId, errorQuery)).donors;
+
+  const validDonorIds = donorData
+    .filter(donor => donor.schemaMetadata.isValid)
+    .map(({ donorId }) => Number(donorId));
+
+  const clinicalErrors: ClinicalErrorsResponseRecord[] = [];
+
+  const invalidDonorIds = errorDonorIds
+    .filter(donorId => !validDonorIds.includes(donorId))
+    .filter((entity, index, array) => array.indexOf(entity) === index);
```

Comment:
I think this can just be 
```typescript
  const invalidDonorIds = errorDonorIds
    .filter(donorId => !validDonorIds.includes(donorId))
```
If there's a second condition here it can be added to the first `filter` function to make it a single iteration.
The current second filter looks like it'll always filter nothing because it's comparing the current index with the index of the current element.. which are equal

Reply:
Updated using reusable duplicates function